includes/class-sensei-course-structure.php

```diff
@@ -84,10 +85,16 @@ public function get( $context = 'view', $no_cache = false ) {
 		$post_status            = $published_lessons_only ? self::PUBLISHED_POST_STATUSES : 'any';
 		$no_module_lessons      = wp_list_pluck( Sensei()->modules->get_none_module_lessons( $this->course_id, $post_status ), 'ID' );
 
+		$add_teacher_name_property = 'edit' === $context && current_user_can( 'manage_options' ) && ( is_admin() || $is_admin_override );
+
+		$add_teacher_name_property && remove_filter( 'get_terms', array( Sensei()->modules, 'append_teacher_name_to_module' ), 70 );
+
 		$modules = $this->get_modules();
-		foreach ( $modules as $module_term ) {
-			$module = $this->prepare_module( $module_term, $post_status );
 
+		$add_teacher_name_property && $context && add_filter( 'get_terms', array( Sensei()->modules, 'append_teacher_name_to_module' ), 70, 3 );
```

Comment:
This logic looks very specific and addresses a special case.

What do you think about using the remove/add filter in the controller (remove the filter before calling this method then add it again)? In that case, you won't need to modify this method at all.

_I have a few more related suggestions in the following comments..._